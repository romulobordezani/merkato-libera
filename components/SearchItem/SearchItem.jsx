import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';

import styles from './SearchItem.scss';
import Price from '../Price';
import ShippingBadge from '../ShippingBadge';

const SearchItem = ({ item }) => {
  return (
    <Link href="/items/[id]" as={`/items/${item.id}`}>
      <div className={styles.SearchItem}>
        <div className={styles.SearchItem_picture}>
          <img src={item.picture} alt={item.title} />
        </div>
        <div className={styles.SearchItem_info}>
          <Price item={item} fontSize="24px" />
          <ShippingBadge item={item} />
          <div className={styles.SearchItem_info_title}>{item.title}</div>
        </div>
        <div className={styles.SearchItem_city}>{item.city}</div>
      </div>
    </Link>
  );
};

SearchItem.propTypes = {
  item: PropTypes.object.isRequired
};

export default SearchItem;
