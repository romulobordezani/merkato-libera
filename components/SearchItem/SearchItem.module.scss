@import "../_SassLib/colors";
@import "../_SassLib/breakpoints";

.SearchItem {

  display: grid;
  grid-template-columns: auto auto 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr;
  column-gap: 1%;

  grid-template-areas:
    "imagBox imagBox infoBox infoBox infoBox infoBox infoBox infoBox cityBox cityBox";

  cursor: pointer;
  border-bottom: 1px solid $light-grey;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
  transition-timing-function: ease-in-out;
  -webkit-transition-timing-function: ease-in-out;

  @include breakpoint($small-portrait) {
    grid-template-rows: auto auto auto;
    grid-template-areas:
      "imagBox imagBox imagBox imagBox imagBox imagBox imagBox imagBox imagBox imagBox"
      "infoBox infoBox infoBox infoBox infoBox infoBox infoBox infoBox infoBox infoBox"
      "cityBox cityBox cityBox cityBox cityBox cityBox cityBox cityBox cityBox cityBox";
    justify-items: start;
  }

  &:hover {
    background-color: lighten( $light-grey, 4% );
  }

  &:last-child {
    border-bottom: none;
  }

  &_picture {
    grid-area: imagBox;
    width: 100%;
    margin: 16px 0;
    outline: none;
    align-items: center;

    @include breakpoint($large) {
      width: 180px;
      height: auto;
      min-height: 180px;
      margin: 16px 0 16px 16px;
    }

    img {
      width: 100%;
      height: auto;
      max-height: 180px;
      object-fit: scale-down;
      border-radius: 4px !important;
    }
  }

  &_info {
    grid-area: infoBox;
    margin: 0 16px 0 0;

    @include breakpoint($small) {
      margin: 0 16px;
    }

    &_title {
      font-weight: 100;
      font-size: 18px;
      word-break: break-word;
    }
  }

  &_city {
    grid-area: cityBox;
    padding: 32px 0;
    font-size: 12px;
    color: $grey;
    @include breakpoint($small) {
      margin: 0 16px;
    }
  }
}
