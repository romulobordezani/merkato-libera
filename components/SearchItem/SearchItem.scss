@import "../_SassLib/colors";

.SearchItem {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: start;
  justify-items: center;
  cursor: pointer;
  padding: 16px;
  border-bottom: 1px solid $light-grey;
  -webkit-transition: background-color 0.3s; /* Safari prior 6.1 */
  transition: background-color 0.3s;
  transition-timing-function: ease-in-out;

  &:hover {
    background-color: lighten( $light-grey, 4% );
  }

  &:last-child {
    border-bottom: none;
  }

  &_col {

    &_1 {
      flex: 10;
    }

    &_2 {
      flex: 80;
    }

    &_3 {
      flex: 10;
    }
  }

  &_picture {
    width: 180px;
    height: 180px;
    outline: none;
    margin: 0 16px;
    align-items: center;
    display: flex;
    background-color: $paper-white;

    img {
      width: 180px;
      height: auto;
      object-fit: cover;
      border-radius: 4px !important;
    }
  }

  &_price {
    font-size: 24px;
    padding: 16px 0;

    &_currency {
      font-weight: 300;
    }
  }

  &_title {
    font-weight: 100;
    font-size: 18px;
  }

  &_city {
    padding: 16px 0;
    font-size: 12px;
    color: $grey;
  }


}
