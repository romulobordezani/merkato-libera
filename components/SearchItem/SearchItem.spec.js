import React from 'react';
import renderer from 'react-test-renderer';

import SearchItem from './SearchItem';

const ItemMock = require('../../__mocks__/Item');

describe('SearchItem', () => {
  it('Snapshot renders default', () => {
    expect(renderer.create(<SearchItem item={ItemMock} />)).toMatchSnapshot();
  });
});
