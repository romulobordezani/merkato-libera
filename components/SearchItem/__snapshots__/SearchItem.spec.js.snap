// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem Snapshot renders default 1`] = `
<div
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div>
    <img
      alt="Botella Deportiva Térmica Reutilizable Pura Vita 500 Ml"
      src="https://mla-s1-p.mlstatic.com/943145-MLA31085352819_062019-O.jpg"
    />
  </div>
  <div>
    <span
      style={
        Object {
          "fontSize": "24px",
        }
      }
    >
      <span>
        $
      </span>
      <span>
        1.800
      </span>
      <span>
        
      </span>
    </span>
    <div>
      Botella Deportiva Térmica Reutilizable Pura Vita 500 Ml
    </div>
  </div>
  <div />
</div>
`;
